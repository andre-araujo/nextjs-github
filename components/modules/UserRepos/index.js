import {
    arrayOf,
    string,
    shape,
    number,
} from 'prop-types';

import UserRepoItem from '../../elements/UserRepoItem';
import Sort from '../../elements/Sort';

import Title from './styles';

const UserRepos = ({
    data,
}) => (
    <section>
        <Title>User repositories</Title>
        <Sort />
        <ul>
            {
                data.map(userRepo => (
                    <UserRepoItem
                        key={userRepo.id}
                        description={userRepo.description}
                        fullname={userRepo.full_name}
                        githubUrl={userRepo.html_url}
                        language={userRepo.language}
                        stars={userRepo.stargazers_count}
                        title={userRepo.name}
                    />
                ))
            }
        </ul>
    </section>
);

UserRepos.defaultProps = {
    data: [],
};

UserRepos.propTypes = {
    data: arrayOf(shape({
        full_name: string.isRequired,
        html_url: string,
        id: number,
        language: string,
        name: string,
        stargazers_count: number,
    })).isRequired,
};

export default UserRepos;
