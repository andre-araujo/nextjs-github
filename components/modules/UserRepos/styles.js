import styled from 'styled-components';

export default styled.h2`
    font-family: AvantGardeDemi;
    font-size: 2rem;
    line-height: 1;
    padding-bottom: .3em;
    margin-bottom: .3em;
    border-bottom: 1px solid #888;
    text-transform: uppercase;
`;
