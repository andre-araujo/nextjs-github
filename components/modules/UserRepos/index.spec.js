import React from 'react';
import { shallow } from 'enzyme';

import UserRepos from './index';

describe('UserRepos component', () => {
    it('should match snapshot with all user repo data', () => {
        expect(shallow(
            <UserRepos
                data={[
                    {
                        html_url: 'html_url',
                        id: 32,
                        language: 'language',
                        name: 'name',
                        stargazers_count: 43,
                        full_name: 'user/repo',
                    },
                ]}
            />,
        )).toMatchSnapshot();
    });
});
