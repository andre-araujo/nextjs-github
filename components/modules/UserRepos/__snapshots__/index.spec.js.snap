// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRepos component should match snapshot with all user repo data 1`] = `
<section>
  <styled.h2>
    User repositories
  </styled.h2>
  <withRoute(Sort) />
  <ul>
    <UserRepoItem
      description={null}
      fullname="user/repo"
      githubUrl="html_url"
      key="32"
      language="language"
      stars={43}
      title="name"
    />
  </ul>
</section>
`;
